include/kafka/Error.h

```diff
@@ -62,5 +64,59 @@ struct ErrorWithDetail
     std::string     detail;
 };
 
+/**
+ * Error type with rich info.
+ */
+class Error
+{
+public:
+    explicit Error(rd_kafka_error_t* error = nullptr): _rkError(error, RkErrorDeleter) {}
```

Comment:
It's the caller's responsibility to delete error?

Reply:
We don't (want to) care about that, since all should be RAII.